### PR TITLE
Rrt star

### PR DESCRIPTION
**Pull request by MichelangeloFantini**
Planning package without RRT star

### CODE REVIEW

File: RRT_star_unbounded.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        edge = v.connect(v_nearest)
+        v.parent = v_nearest
+        v.parent = v_nearest
+        v.edge = edge
+        q_min = v_nearest.coord
+
+        for vertex in neighborhood:
+            point = vertex.coord
+            dist = np.linalg.norm(point -q_min)
+            if dist== 0:
```
Comment (adamheins) [inline RRT_star_unbounded.py]: Don't check for equality with floating point numbers: do something like `dist <= 1e-8`. We need a threshold due to floating point rounding error.

File: RRT_star.py [hunk 2]
```diff
@@ -37,119 +43,149 @@ def connection(self, va, vb):
...
-        q_nearest = v_nearest.coord
+    
+    def query(
+        self,
+        start,
+        goal,
+        n=1000,
+        min_edge_len=0.5,
+        max_edge_len=1,
+        niu = 0.8
```
Comment (adamheins) [inline RRT_star.py]: What is the purpose of the niu variable? Can you add a comment and/or change the name to be more descriptive?

File: RRT_star.py [hunk 3]
```diff
@@ -37,119 +43,149 @@ def connection(self, va, vb):
...
+        start_time = time.time()
+        samples = []
+        prev = 0
+        while new_size<n:
+            # if prev !=new_size:
+            #     print(new_size)
+            #     prev = new_size
+            # TODO maybe change min and max edge len based on how many points have been taken?
+            # add radius with minimum path
+            if len(samples) == 0:
```
Comment (adamheins) [inline RRT_star.py]: What is the different between the case when `len(samples) == 0` vs `len(samples) > 0`? Can you comment what is going on here?

File: RRT_star.py [hunk 4]
```diff
@@ -1,25 +1,31 @@
...
         cost = 0
-        vb = tuple(vb)
-        for key in self.cost_dic.keys():
-            if key == vb: 
-                cost = self.cost_dic[vb]
-            # elif key == va:
-            #     cost_a = self.cost_dic[vb]
+        parent = vb.parent
+        while vb.parent != None:
+            cost += self.connection(vb.coord,parent.coord)
```
Comment (adamheins) [inline RRT_star.py]: Can you change the name of the `connection` method to `connection_cost` to make it really clear what it does?

Comment (MichelangeloFantini) [reply RRT_star.py]: Sounds good!

File: RRT_star.py (hunk 3 above)
Comment (MichelangeloFantini) [reply RRT_star.py]: samples is a list that is actually used mostly in the unbounded RRT star although I included it here for consistency. It stores all the points that have been sampled and are safe to add to the tree. When samples is empty the code first looks for a new point and then tries to add it. If samples is not empty then the workspace is not sampled as there are already points that can be added to the tree.

File: RRT_star.py (hunk 2 above)
Comment (MichelangeloFantini) [reply RRT_star.py]: niu is the maximum distance that we are willing to shift the sampled point toward the closest point in the tree. it's a constant that in research papers is usually denoted as niu. I don't know much about it but I can definitely research more

**Review by adamheins — APPROVED** [review]
Made some changes to organize and streamline some things. Otherwise looks good.

**Pull request merged**